discovery-provider/src/api/v1/users.py

```diff
@@ -837,26 +858,39 @@ def get(self, id):
 
 @full_ns.route("/content_node/<string:replica_type>", doc=False)
 class UsersByContentNode(Resource):
-    @full_ns.marshal_with(users_by_content_node_response)
-    # @cache(ttl_sec=30)
-    def get(self, replica_type):
-        """New route to call get_users_cnode with replica_type param (only consumed by content node)
-        - Leaving `/users/creator_node` above untouched for backwards-compatibility
+    @ns.doc(
+        id="""Get Users By Replica Type for Content Node""",
+        description="""
+        (Only consumed by Content Node) Gets users that have a given Content Node endpoint as
+        their primary, secondary, or either (depending on the replica_type passed).
 
         Response = array of objects of schema {
             user_id, wallet, primary, secondary1, secondary2, primarySpId, secondary1SpID, secondary2SpID
         }
-        """
+        """,
+        responses={200: "Success", 400: "Bad request", 500: "Server error"},
+    )
+    @full_ns.marshal_with(users_by_content_node_response)
+    # @cache(ttl_sec=30)
+    def get(self, replica_type):
         args = users_by_content_node_route_parser.parse_args()
 
-        cnode_url = args.get("creator_node_endpoint")
+        cnode_url = args.get("content_node_endpoint") or args.get(
+            "creator_node_endpoint"
+        )
+        prev_user_id = args.get("prev_user_id") or 0
+        max_users = args.get("max_users") or 100000
 
         if replica_type == "primary":
-            users = get_users_cnode(cnode_url, ReplicaType.PRIMARY)
+            users = get_users_cnode(
+                cnode_url, ReplicaType.PRIMARY, prev_user_id, max_users
+            )
         elif replica_type == "secondary":
-            users = get_users_cnode(cnode_url, ReplicaType.SECONDARY)
+            users = get_users_cnode(
+                cnode_url, ReplicaType.SECONDARY, prev_user_id, max_users
+            )
         else:
-            users = get_users_cnode(cnode_url, ReplicaType.ALL)
+            users = get_users_cnode(cnode_url, ReplicaType.ALL, prev_user_id, max_users)
```

Comment:
im assuming that the prev_user_id here is used as the offset cus this returns users in ascending order? and max_users is the limit? if so, let's add a comment describing that :)